src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,173 @@
+import * as registry from "dijit/registry";
+
+import { SwipeHandler } from "./SwipeHandler";
+import "./ui/TabSwipe.scss";
+
+export interface TabContainer extends mxui.widget._WidgetBase {
+    declaredClass: "mxui.widget.TabContainer";
+    showTab: (tab: TabPane) => void;
+    focusIndex: number;
+    onShowTab: (callback: () => void ) => void;
+    onHideTab: (callback: () => void ) => void;
+    tabSwipeId: string;
+    validator: null;
+    _active: TabPane;
+    _tabList: HTMLElement;
+    _tabPanes: TabPane[];
+    _tabContent: TabContent;
+    _clickHandler: null;
+}
+
+export interface TabContent extends HTMLElement {
+    declaredClass: "mxui.widget.TabContent";
+}
+
+export interface TabPane extends mxui.widget._WidgetBase {
+    _hidden: boolean;
+    _parsed: boolean;
+    _visible: boolean;
+    container: TabContainer;
+    showTab: (tab: TabPane) => void;
+    show: () => void;
+    hideTab: () => void;
+    index: number;
+    visibilityIndex: number;
+}
+
+class TabSwipe extends WidgetBase {
+    targetName: string;
+    tabStyle: "tabs"| "indicators";
+    lazyLoad: boolean;
+
+    private targetWidget?: TabContainer;
+    private targetNode: HTMLElement;
+    private swipeHandler: SwipeHandler;
+    private tabContentClass: string;
+    private swipeClass: string;
+    private targetWidgetType: string;
+
+    postCreate() {
+        this.tabContentClass = ".mx-tabcontainer-content";
+        this.swipeClass = "widget-tab-swipe";
+        this.targetWidgetType = "mxui.widget.TabContainer";
+        this.targetNode = this.findTargetNode(this.targetName);
+        this.targetWidget = this.targetNode ? registry.byNode(this.targetNode) : null;
+
+        if (!this.targetWidget) {
+            this.showError(`Tab swipe configuration error: unable to find a target with the name ${this.targetName}`);
+        } else if (this.checkCompatibility(this.targetWidget)) {
+            this.targetWidget.tabSwipeId = this.id;
+            this.targetNode.classList.add(this.swipeClass);
+        }
+    }
+
+    update(_contextObject: mendix.lib.MxObject, callback?: () => void) {
+        if (this.targetWidget) {
+            this.applyIndicatorStyles(this.targetWidget);
+            this.initializeSwipe(this.targetWidget);
+            this.setupEvents(this.targetWidget);
+        }
+
+        if (callback) {
+            callback();
+        }
+    }
+
+    uninitialize(): boolean {
+        if (this.swipeHandler) {
+            this.swipeHandler.destroy();
+        }
+
+        return true;
+    }
+
+    private findTargetNode(name: string): HTMLElement {
+        let queryNode = this.domNode.parentNode as Element;
+        let targetNode: HTMLElement | null = null;
+        while (!targetNode) {
+            targetNode = queryNode.querySelector(".mx-name-" + name) as HTMLElement;
+            if (window.document.isEqualNode(queryNode)) { break; }
+            queryNode = queryNode.parentNode as HTMLElement;
```

Comment:
Should we check for off screen rendering as wel?
If it is rendering the widget off screen it wil pick a the tab from the previous page.

``` js
if (queryNode.classList.contains("mx-incubator") || queryNode.classList.contains("mx-offscreen")) { break; }